backend/pkg/httpserver/get_saved_search.go

```diff
@@ -16,25 +16,44 @@ package httpserver
 
 import (
 	"context"
-	"fmt"
+	"errors"
+	"log/slog"
+	"net/http"
 
+	"github.com/GoogleChrome/webstatus.dev/lib/gcpspanner/spanneradapters/backendtypes"
 	"github.com/GoogleChrome/webstatus.dev/lib/gen/openapi/backend"
+	"github.com/GoogleChrome/webstatus.dev/lib/httpmiddlewares"
 )
 
 // GetSavedSearch implements backend.StrictServerInterface.
 // nolint:ireturn, revive // Expected ireturn for openapi generation.
 func (s *Server) GetSavedSearch(
-	_ context.Context, req backend.GetSavedSearchRequestObject) (
+	ctx context.Context, req backend.GetSavedSearchRequestObject) (
 	backend.GetSavedSearchResponseObject, error) {
-	savedSearches := getSavedSearches()
-	for _, search := range savedSearches {
-		if req.SearchId == search.Id {
-			return backend.GetSavedSearch200JSONResponse(search), nil
+	// At this point, the user should be authenticated and in the context.
+	// If for some reason the user is not in the context, treat it as an unauthenticated user
+	var userID *string
+	user, found := httpmiddlewares.AuthenticatedUserFromContext(ctx)
+	if found {
```

Comment:
Should it return here if a user is not `found`? I assume there are no saved searches for unauthenticated users

Reply:
In this case, no. The unauthenticated case still should allow the response to go through. For example, if I create a public saved search, you should be able to see it so that you can bookmark it too.  (This is meant so that a team can create a saved search and share it amongst the team or publicly.)

The authenticated case just returns extra information regarding if you have already bookmarked it or not.

https://github.com/GoogleChrome/webstatus.dev/blob/10c5f9dea62e0d3a19da7f2a17346d2b00617e74/lib/gcpspanner/get_user_saved_search.go#L111-L130

https://github.com/GoogleChrome/webstatus.dev/blob/10c5f9dea62e0d3a19da7f2a17346d2b00617e74/lib/gcpspanner/get_user_saved_search.go#L94

Reply:
Gotcha. Thanks!